src/util/virutil.c

```diff
@@ -277,6 +277,33 @@ virParseVersionString(const char *str, unsigned long *version,
     return 0;
 }
 
+int
+virCHVersionString(const char *str, unsigned long *version,
+                      bool allowMissing)
+{
+    unsigned int major, minor = 0, micro = 0;
+
+    if (strncmp(str, "cloud-hypervisor v", 18) == 0) {
+        // check for regular 'cloud-hypervisor --version' output
+        // for ex: cloud-hypervisor v32.0.131
+        // for ex: cloud-hypervisor v33.0-104-ge0e3779e-dirty
+        sscanf(str, "cloud-hypervisor v%d.%d.%d", &major, &minor, &micro);
```

Comment:
Perhaps should still use `virParseVersionString` to retain the behavior?

Thanks

Reply:
`virParseVersionString` might not work with **cloud-hypervisor v33.0-104-ge0e3779e-dirty**

This is the format we got for main branch of public CLH.